### PR TITLE
popular movie page api with redux saga

### PR DESCRIPTION
**Pull request by WojciechBylica**
at this step navigation is not working good - i think that its because temporary we have two sources of data, or its the problem with "page" element. But think that next step - moving all api data to reducers will fix the problem.

### CODE REVIEW

File: src/features/MovieList/index.js [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
+    const dispatch = useDispatch();
+    const status = useSelector(selectPopularMoviesStatus);
+
+    const page = useQueryParameter(searchQueryParamName) || "1";
+
+    useEffect(() => {
+        dispatch(fetchPopularMoviesLoading(page));
+    }, [dispatch, page]);
+
     const APIData = useAPIData();
```
Comment (KarolCiesluk) [inline src/features/MovieList/index.js:25]: are you still using useAPIData hook here?? sure about it? :)

File: src/features/MovieList/popularMoviesSlice.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const popularMoviesSlice = createSlice({
+    name: "popularMovies",
+    initialState: {
+        status: "initial",
+        popularPeopleData: null,
```
Comment (KarolCiesluk) [inline src/features/MovieList/popularMoviesSlice.js:7]: should be popularMoviesData

File: src/features/MovieList/popularMoviesSlice.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    reducers: {
+        fetchPopularMoviesLoading: () => ({
+            status: "loading",
+        }),
+        fetchPopularMoviesSuccess: (_, { payload: popularMoviesData }) => ({
```
Comment (KarolCiesluk) [inline src/features/MovieList/popularMoviesSlice.js:13]: fetchPopularMoviesSuccess

File: src/features/MovieList/popularMoviesSlice.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const popularMoviesSlice = createSlice({
+    name: "popularMovies",
+    initialState: {
+        status: "initial",
+        popularPeopleData: null,
+    },
+    reducers: {
+        fetchPopularMoviesLoading: () => ({
```
Comment (KarolCiesluk) [inline src/features/MovieList/popularMoviesSlice.js:10]: I would simply call it `fetchPopularMovies`. The same way Chris showed it in ToDoList.

File: src/features/PersonList/popularPeopleApi.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import axios from "axios";
 
-const APIKey = "api_key=3af561a12389e6d632bf79207cb88b6c";
-const pathAPI = "https://api.themoviedb.org/3/";
+import { APIKey } from "../../APIKey";
+import { pathAPI } from "../../pathAPI";
```
Comment (KarolCiesluk) [inline src/features/PersonList/popularPeopleApi.js:4]: I've put them both in common/movieAPI on branch features/MovieDetails so we are gonna have a conflict. I don't mind your way. We just have to decide what we do and do the same :)

**Review by KarolCiesluk — CHANGES_REQUESTED** [review]
Great job! I hope we won't have too many conflicts 😄

**Review by KarolCiesluk — APPROVED** [review]

File: src/features/MovieList/index.js [hunk 6]
```diff
@@ -1,28 +1,42 @@
...
+        switch (status) {
             case "loading":
                 return < Loader />;
             case "success":
                 return (
                     <>
                         <Wrapper>
                             <MovieTiles title="Popular movies"
-                                popularMoviesData={APIData.popularMoviesData.data}
                                 movieData={APIData.movieData.data}
```
Comment (KarolCiesluk) [inline src/features/MovieList/index.js:36]: if you want to use redux you don't need props here.

File: src/features/Tiles/MovieTiles/index.js [hunk 7]
```diff
@@ -12,14 +15,13 @@ import {
 } from "./styled";
 import { Rate } from "../../Rate";
 
+const MovieTiles = ({ title, movieData }) => {
```
Comment (KarolCiesluk) [inline src/features/Tiles/MovieTiles/index.js:18]: movieData is probably not needed here. You can get all the necessary info from popularMoviesData i suppose... 🤔

File: src/features/MovieList/index.js (hunk 6 above)
Comment (KarolCiesluk) [reply src/features/MovieList/index.js:36]: I mean you probably still need `title="Popular movies"`

**Review by KarolCiesluk — CHANGES_REQUESTED** [review]
make sure to have a look on branch common/Pagination before you merge :)

File: src/features/MovieList/popularMoviesSlice.js (hunk 2 above)
Comment (WojciechBylica) [reply src/features/MovieList/popularMoviesSlice.js:7]: yes

File: src/features/PersonList/popularPeopleApi.js (hunk 5 above)
Comment (WojciechBylica) [reply src/features/PersonList/popularPeopleApi.js:4]: ok. i dont change it now, we`ll decide after merging both branches

File: src/features/MovieList/index.js (hunk 6 above)
Comment (WojciechBylica) [reply src/features/MovieList/index.js:36]: yes, but after adding next api data - we need genres. its to be changed in next step

File: src/features/MovieList/popularMoviesSlice.js (hunk 4 above)
Comment (WojciechBylica) [reply src/features/MovieList/popularMoviesSlice.js:10]: please leave it for now.  as this reducer only sets status to loading, it should has loading in name.

**Comment by WojciechBylica** [comment]
I added second - final step. Data is provided by reducers, saga is taking care about side effects, and genres are correct!

Still missing styling for genres, but its another task which I put to our Trello

File: src/features/Tiles/MovieTiles/index.js [hunk 8]
```diff
@@ -44,7 +45,7 @@ const MovieTiles = ({ title, popularMoviesData, movieData }) => {
                                 )}
 
                                 <Tags>
-                                    {genres.map(tag => <Tag key={`genres-${tag.name}`}>{tag.name}</Tag>)}
+                                    {moviesList[index].genre_ids.map(movie => <Tag key={`genres-${movie}`}>{moviesGenres.find(({ id }) => id === movie).name}</Tag>)}
```
Comment (KarolCiesluk) [inline src/features/Tiles/MovieTiles/index.js:48]: This line is too long on my screen so maybe needs some formatting.

I'm also a bit unsure about the names you use here.
`moviesList[index]`: I don't think you need index here. Just call it movie like in the code above.
`movie`: It's not really a movie but genreID
There are many ways to write it but I would do something like that:
`{movie.genre_ids.map(genreID => <Tag key={`genres-${genreID}`}>{moviesGenres.find(({ id }) => id === genreID).name}</Tag>)}`
or maybe `{genreID}` is enough in key...

**Review by KarolCiesluk — APPROVED** [review]
Really great job! 👍  Everything works very well. 

I'm only wondering about the way we fetch genres. The list of genres never changes but now we fetch it every time we click on movies in navigation and every time we use pagination. That seems to be a little bit unproductive.
I guess we will also need to fetch configuration (https://developers.themoviedb.org/3/configuration/get-api-configuration) at some point intead of hard coding it. Maybe both configuration and genres should be fetched together only once at the beginning of the app for example in App.js... but would we need a new slice then?? 🤔 

I am not really sure what's the best way of dealing with it but it's something to talk about 😄 
For now it's not so crucial so I approve your solution! 💪 

P.S. Are you going to add the updated Pagination from branch common/Pagination to this branch or you prefer to merge it separately?

File: src/features/MovieList/index.js (hunk 6 above)
Comment (KarolCiesluk) [reply src/features/MovieList/index.js]: 👍

File: src/features/Tiles/MovieTiles/index.js (hunk 8 above)
Comment (WojciechBylica) [reply src/features/Tiles/MovieTiles/index.js]: true. I`ve forgotten that its already in map and we can use "movie" to get inside.

**Comment by WojciechBylica** [comment]
I changed naming as Karol suggested and resolved conflicts.

**Pull request merged**